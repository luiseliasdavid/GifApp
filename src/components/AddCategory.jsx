import { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ agregarElement }) => {
  const [inputValue, setInputValue] = useState('')

  const onInputChange = ({ target }) => {
    setInputValue(target.value)
  }

  const onSubmit = (event) => {
    event.preventDefault()
    if (inputValue.trim().length <= 1) return
    agregarElement(inputValue.trim())
    setInputValue('')
  }

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  agregarElement: PropTypes.func.isRequired,
}
