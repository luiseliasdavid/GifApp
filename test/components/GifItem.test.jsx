import { render,screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Probando componente <GiftItem/>', () => {

    const title = "some title"
    const url = 'https://some-url/url.jpg'

  test('should match with snapshot', () => {
    
    const {container}= render(<GifItem title={title} url={url} />)
    expect(container).toMatchSnapshot()
  })

  test('should  render img with URL and ALT indicated', () => {
    
    render(<GifItem title= {title} url={url}/>)
    
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  })

  test('should show the title in the document', () => {
    render(<GifItem title= {title} url={url}/>)
    expect(screen.getByText(title)).toBeTruthy()
  })
  
  
  
})
