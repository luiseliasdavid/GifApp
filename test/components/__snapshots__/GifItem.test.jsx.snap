// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente <GiftItem/> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="some title"
      src="https://some-url/url.jpg"
    />
    <p>
       
      some title
       
    </p>
  </div>
</div>
`;
